python/kubeflow/trainer/api/trainer_client.py

```diff
@@ -13,587 +13,125 @@
-
-    def list_runtimes(self) -> List[types.Runtime]:
+        Args:
+            backend_type: name of the backend to be used. default is kubernetes.
+            backend_config: backend configuration. default is None.
+        """
+        backend = self.__init_backend(backend_type, backend_config)
+        self.__backend = backend
+
+    def __init_backend(self, backendtype: str, backendconfig: BackendConfig):
+        backend = TRAINER_BACKEND_REGISTRY.get(backendtype.lower())
+        if not backend:
+            raise ValueError("Unknown backend type '{}'".format(backendtype))
+        # load the backend class
+        backend_cls = backend.get("backend_cls")
+        # check if backend configuration is present
+        if not backendconfig:
+            backendconfig = backend.get("config_cls")()
+        # initialize the backend class with the user provided config
+        return backend_cls(cfg=backendconfig)
+
+    def list_runtimes(self):
         """List of the available Runtimes.
 
-        Returns:
-            List[Runtime]: List of available training runtimes.
-                If no runtimes exist, an empty list is returned.
+            Returns:
+                List[Runtime]: List of available training runtimes.
+                    If no runtimes exist, an empty list is returned.
 
-        Raises:
-            TimeoutError: Timeout to list Runtimes.
-            RuntimeError: Failed to list Runtimes.
+            Raises:
+                TimeoutError: Timeout to list Runtimes.
+                RuntimeError: Failed to list Runtimes.
         """
+        return self.__backend.list_runtimes()
 
-        result = []
-        try:
-            thread = self.custom_api.list_cluster_custom_object(
-                constants.GROUP,
-                constants.VERSION,
-                constants.CLUSTER_TRAINING_RUNTIME_PLURAL,
-                async_req=True,
-            )
-
-            runtime_list = models.TrainerV1alpha1ClusterTrainingRuntimeList.from_dict(
-                thread.get(constants.DEFAULT_TIMEOUT)
-            )
-
-            if not runtime_list:
-                return result
+    def get_runtime(self, name: str):
+        return self.__backend.get_runtime(name=name)
 
-            for runtime in runtime_list.items:
-                result.append(self.__get_runtime_from_crd(runtime))
+    def list_jobs(self, runtime: Optional[types.Runtime] = None):
+        """List of all TrainJobs.
 
-        except multiprocessing.TimeoutError:
-            raise TimeoutError(
-                f"Timeout to list {constants.CLUSTER_TRAINING_RUNTIME_KIND}s "
-                f"in namespace: {self.namespace}"
-            )
-        except Exception:
-            raise RuntimeError(
-                f"Failed to list {constants.CLUSTER_TRAINING_RUNTIME_KIND}s "
-                f"in namespace: {self.namespace}"
-            )
+            Returns:
+                List[TrainerV1alpha1TrainJob]: List of created TrainJobs.
```

Comment:
It looks like return type is incorrect in doc-string.
```suggestion
                List[TrainJob]: List of created TrainJobs.
```